### PR TITLE
fix renaming

### PR DESCRIPTION
**Pull request by gronke**
fixes renaming jails

### CODE REVIEW

File: libiocage/lib/JailConfig.py [hunk 1]
```diff
@@ -37,6 +37,16 @@
 import libiocage.lib.Jail
 
 
+class SpecialProperties(dict):
```
Comment (igalic) [inline libiocage/lib/JailConfig.py]: this is name is a red flag :D

File: libiocage/cli/rename.py [hunk 2]
```diff
@@ -39,7 +39,7 @@ def cli(ctx, jail, name):
     """
 
     logger = ctx.parent.logger
-    try :
+    try:
```
Comment (igalic) [inline libiocage/cli/rename.py]: while you're touching these functions, you might as well add `-> None` to the the functions (and perhaps _other_ type signatures, too :)

File: libiocage/lib/Jail.py [hunk 3]
```diff
@@ -409,11 +414,18 @@ def rename(self, new_name: str):
...
+            dataset.rename(new_dataset_name)
+            self.logger.verbose(
+                f"Dataset {current_dataset_name} renamed to {new_dataset_name}"
+            )
         except:
-            self.config["name"] = current_id
+            self.config["id"] = current_id
             raise
 
     def _force_stop(self):
```
Comment (igalic) [inline libiocage/lib/Jail.py]: `-> bool`

File: libiocage/lib/Jail.py [hunk 4]
```diff
@@ -409,11 +414,18 @@ def rename(self, new_name: str):
...
         try:
-            dataset.rename(self.dataset_name)
+            new_dataset_name = self._dataset_name_from_jail_name
+            dataset.rename(new_dataset_name)
+            self.logger.verbose(
+                f"Dataset {current_dataset_name} renamed to {new_dataset_name}"
+            )
         except:
-            self.config["name"] = current_id
+            self.config["id"] = current_id
```
Comment (igalic) [inline libiocage/lib/Jail.py]: what's up with the moving back and forth between `id` and `name`

Comment (gronke) [reply libiocage/lib/Jail.py]: It's always ID. Name should always point to ID instead.

Comment (igalic) [reply libiocage/lib/Jail.py]: aye

**Comment by gronke** [comment]
Closing in favor of #89

**Pull request closed without merging**